x-pack/packages/kbn-elastic-assistant/impl/assistant/index.tsx

```diff
@@ -141,181 +102,73 @@ const AssistantComponent: React.FC<Props> = ({
-          find(conversations, ['title', getLastConversationId(WELCOME_CONVERSATION_TITLE)]);
-
-        if (deepEqual(prev, nextConversation)) return prev;
-
-        const conversationToReturn =
-          (nextConversation &&
-            conversations[
-              nextConversation?.id !== '' ? nextConversation?.id : nextConversation?.title
-            ]) ??
-          conversations[WELCOME_CONVERSATION_TITLE] ??
-          getDefaultConversation({ cTitle: WELCOME_CONVERSATION_TITLE });
-
-        // updated selected system prompt
-        setEditingSystemPromptId(
-          getDefaultSystemPrompt({
-            allSystemPrompts,
-            conversation: conversationToReturn,
-          })?.id
-        );
-        if (
-          prev &&
-          prev.id === conversationToReturn.id &&
-          // if the conversation id has not changed and the previous conversation has more messages
-          // it is because the local conversation has a readable stream running
-          // and it has not yet been persisted to the stored conversation
-          prev.messages.length > conversationToReturn.messages.length
-        ) {
-          return {
-            ...conversationToReturn,
-            messages: prev.messages,
-          };
-        }
-        return conversationToReturn;
-      });
-    }
-  }, [
+  const {
+    currentConversation,
+    currentSystemPromptId,
+    handleCreateConversation,
+    handleOnConversationDeleted,
+    handleOnConversationSelected,
+    handleOnSystemPromptSelectionChange,
+    refetchCurrentConversation,
+    setCurrentConversation,
+    setCurrentSystemPromptId,
+  } = useCurrentConversation({
     allSystemPrompts,
-    areConnectorsFetched,
-    conversationTitle,
     conversations,
-    conversationsLoaded,
-    currentConversationId,
-    getDefaultConversation,
-    getLastConversationId,
-    isAssistantEnabled,
-  ]);
+    defaultConnector,
+    refetchCurrentUserConversations,
+    conversationId: getLastConversationId(conversationTitle),
+    mayUpdateConversations:
+      isFetchedConnectors &&
+      isFetchedCurrentUserConversations &&
+      isFetchedPrompts &&
+      Object.keys(conversations).length > 0,
+  });
 
   // Welcome setup state
-  const isWelcomeSetup = useMemo(() => {
-    // if any conversation has a connector id, we're not in welcome set up
-    return Object.keys(conversations).some(
-      (conversation) => conversations[conversation]?.apiConfig?.connectorId != null
-    )
-      ? false
-      : (connectors?.length ?? 0) === 0;
-  }, [connectors?.length, conversations]);
-  const isDisabled = isWelcomeSetup || !isAssistantEnabled;
-
-  // Welcome conversation is a special 'setup' case when no connector exists, mostly extracted to `ConnectorSetup` component,
-  // but currently a bit of state is littered throughout the assistant component. TODO: clean up/isolate this state
-  const blockBotConversation = useMemo(
```

Comment:
All this block bot stuff is now `UpgradeLicenseCallToAction`. All of this type of logic (what to show in the body: upgrade cta, welcome setup, empty convo, or messages) is now in `assistant_body/index.tsx`